### PR TITLE
Backend/refactor/move dashboard apis to dsl

### PR DESCRIPTION
**Pull request by roman-bodavskiy**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Steps to move provider-dashboard apis to DSL:

1.  Create/update spec file here `/dashboard/CommonAPIs/ProviderPlatform/spec/API/<module_name>.yaml`
2.  Generate code using `, run-generator  -- --apply-hint`. Then most likely you will follow compiler messages
3. Remove existing handmade code from these modules:
-  **common apis**:` /dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/<module_name>.hs`
you can add reexport in `/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/<module_name>.hs`:
` import API.Types.ProviderPlatform.<module_name> as Reexport`
or remove `/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/<module_name>.hs` if all code moved to DSL

- **bpp**:` /provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/<folder_name>/<module_name>.hs`

- **dashboard client**: `/dashboard/provider-dashboard/src/ProviderPlatformClient/DynamicOfferDriver/<folder_name>.hs`

- **dashboard**: `dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/<module_name>.hs`

4. add your `module_name` here if its not added yet:

- **bpp**: `/provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/<folder_name>.hs`

- **dashboard**: `/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver.hs`

- **dashboard client**: `/dashboard/provider-dashboard/src/ProviderPlatformClient/DynamicOfferDriver/<folder_name>.hs`

5. check handler name and params order, they could change while generation
    `Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/<folder_name>/<module_name>.hs`
    in future we will keep params order

6. fix tests build `Backend/test/src/Mobility/ARDU/APICalls.hs`

example;
`folder_name = Management` (`Operations` on dashboard side)
`module_name = Ride`

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?
tested locally

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [x] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/MerchantServiceConfigExtra.hs [hunk 1]
```diff
@@ -51,5 +51,10 @@ upsertMerchantServiceConfig merchantServiceConfig opCity = do
         [ Se.Set BeamMSC.configJSON configJSON,
           Se.Set BeamMSC.updatedAt now
         ]
-        [Se.Is BeamMSC.merchantId (Se.Eq $ getId merchantServiceConfig.merchantId)]
+        [ Se.And
+            [ Se.Is BeamMSC.merchantId $ Se.Eq $ getId merchantServiceConfig.merchantId,
+              Se.Is BeamMSC.merchantOperatingCityId $ Se.Eq $ Just $ getId opCity,
+              Se.Is BeamMSC.serviceName $ Se.Eq _serviceName
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/MerchantServiceConfigExtra.hs:57]: it appears while testing that dashboard apis change all merchants service configs for each `ServiceName`, that's why this condition added

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/MerchantServiceConfigExtra.hs [hunk 2]
```diff
@@ -71,7 +71,8 @@ upsertMerchantServiceConfig merchantServiceConfig = do
         [Se.Set BeamMSC.configJSON configJSON, Se.Set BeamMSC.updatedAt now]
         [ Se.And
             [ Se.Is BeamMSC.merchantId $ Se.Eq $ getId merchantServiceConfig.merchantId,
-              Se.Is BeamMSC.merchantOperatingCityId $ Se.Eq $ getId merchantServiceConfig.merchantOperatingCityId
+              Se.Is BeamMSC.merchantOperatingCityId $ Se.Eq $ getId merchantServiceConfig.merchantOperatingCityId,
+              Se.Is BeamMSC.serviceName $ Se.Eq _serviceName
```
Comment (roman-bodavskiy) [inline Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/MerchantServiceConfigExtra.hs:75]: it appears while testing that dashboard apis change all merchants service configs for each ServiceName, that's why this condition added

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/spec/Storage/DriverInformation.yaml [hunk 3]
```diff
@@ -189,6 +189,7 @@ DriverInformation:
     updateTollRelatedIssueCount:
       kvFunction: updateOneWithKV
       params : [tollRelatedIssueCount]
+      where: driverId
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/spec/Storage/DriverInformation.yaml:192]: added this fix for correct generation

File: Backend/app/rider-platform/rider-app/Main/src-read-only/Storage/Beam/Ride.hs [hunk 4]
```diff
@@ -79,6 +79,6 @@ instance B.Table RideT where
 
 type Ride = RideT Identity
 
-$(enableKVPG ''RideT ['id] [['bookingId], ['bppRideId]])
+$(enableKVPG ''RideT ['id] [['bookingId], ['bppRideId], ['shortId]])
```
Comment (roman-bodavskiy) [inline Backend/app/rider-platform/rider-app/Main/src-read-only/Storage/Beam/Ride.hs:82]: autogenrated

**Review by hkmangla — APPROVED** [review]

**Pull request merged**